packages/checkpoint/data_stream/firewall/elasticsearch/ingest_pipeline/default.yml

```
@@ -1076,6 +1076,30 @@ processors:
       formats: ["ISO8601", "UNIX"]
       target_field: event.end
       if: "ctx.checkpoint?.lastupdatetime != null"
+  - date:
+      field: checkpoint.last_hit_time
+      formats: ["ISO8601", "UNIX"]
+      target_field: event.end
+      if: "ctx.checkpoint?.last_hit_time != null"
+  - date:
+      field: checkpoint.creation_time
+      formats: ["ISO8601", "UNIX"]
+      target_field: event.start
+      if: "ctx.checkpoint?.creation_time != null"
+  - rename:
+      field: checkpoint.duration
+      target_field: event.duration
+      ignore_missing: true
+  - convert:
+      tag: convert_event_duration_to_integer
+      field: event.duration
+      type: integer
+      ignore_missing: true
+  - script:
+      tag: event_duration_conversion_to_long
+      lang: painless
+      source: ctx.event.duration = ctx.event.duration * 1000000000L
+      if: ctx?.event?.duration != null
```

```suggestion
      if: ctx.event?.duration != null
```

ctx can never be null in this context.